### PR TITLE
TST: GitHub actions

### PR DESCRIPTION
**Pull request by jklenzing**
Addresses pysat/pysat#777.  Replaces #44

- Migrates CI tests to Github Actions
- Updates requirements files and setup.cfg
- Updates readme file now that pysat 3.0 is released

### CODE REVIEW

**Comment by jklenzing** [comment]
The GPS instrument currently fails to load.  Looking into to this more, but putting this up to link to in issues.

**Comment by jklenzing** [comment]
Added a version cap for xarray and documented the bug in #46.  Since the latest xarray is not yet in conda, the Travis CI tests do not catch this.

The one failing test is a result of download issues, likely a server hiccup.  Opening for review.

**Comment by jklenzing** [comment]
Added the macos tests for completeness.

File: requirements.txt [hunk 1]
```diff
@@ -1,5 +1,6 @@
 h5py
+madrigalWeb
 numpy
 pandas
 pysat>=3.0
-xarray
+xarray<0.18
```
Comment (aburrell) [inline requirements.txt:6]: Do we have an issue capturing the changes needed to remove this cap?

File: setup.cfg [hunk 2]
```diff
@@ -41,18 +41,17 @@ setup_requires = setuptools >= 38.6; pip >= 10
...
 [coverage:report]
-omit = */instruments/templates/
+omit = */instruments/templates/*
 
 [flake8]
 max-line-length = 80
+ignore =
+  W503
+exclude =
+  docs/conf.py
```
Comment (aburrell) [inline setup.cfg:54]: Need an issue for this, as well.

File: requirements.txt (hunk 1 above)
Comment (aburrell) [reply requirements.txt:6]: Yes: #46.

File: setup.cfg (hunk 2 above)
Comment (jklenzing) [reply setup.cfg:54]: Added #48

**Comment by jklenzing** [comment]
@aburrell, I'm running into an issue where the remote server sometimes fails to send back a file.  Sadly, github actions only has the option to rerun all jobs rather than a single failing job, which means it is highly likely that the server will fail again.  Ran the latest versions of this several times last night, with the best case where only one job would fail.

This is on GA's radar, but no word as to when it will be fixed:  https://github.com/actions/runner/issues/432

**Comment by aburrell** [comment]
> This is on GA's radar, but no word as to when it will be fixed: actions/runner#432

Seems like the best course of action, for the moment, is to stick with Travis for this package.  We can either close this PR and put an issue in specifying the branch with the work, or convert this to a draft and tag it as Future.  Thoughts?

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install standard dependencies
+      run: |
+        python -m pip install --upgrade pip
```
Comment (aburrell) [inline .github/workflows/main.yml:33]:
```suggestion
```

File: setup.cfg [hunk 4]
```diff
@@ -41,15 +41,10 @@ setup_requires = setuptools >= 38.6; pip >= 10
 include_package_data = True
 zip_safe = False
 packages = find:
-install_requires = h5py
-		   madrigalWeb
-		   numpy
-		   pandas
-		   pysat
-		   xarray
+install_requires = file: requirements.txt
```
Comment (aburrell) [inline setup.cfg:44]: This doesn't work for the install_requires, need to revert.

**Review by aburrell — APPROVED** [review]

**Pull request merged**